File: backend/src/main/java/com/demo/MoneyMap/service/AlertService.java
package com.demo.MoneyMap.service;

import com.demo.MoneyMap.dto.request.AlertStatusUpdateDTO;
import com.demo.MoneyMap.dto.response.AlertResponseDTO;
import com.demo.MoneyMap.dto.response.PagedResponseDTO;
import com.demo.MoneyMap.entity.enums.AlertSeverity;
import com.demo.MoneyMap.entity.enums.AlertStatus;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Map;

/**
 * Service interface for alert management operations.
 */
public interface AlertService {

    /**
     * Get alert by ID.
     */
    AlertResponseDTO getAlertById(Long id);

    /**
     * Get alert by ID with full details.
     */
    AlertResponseDTO getAlertWithDetails(Long id);

    /**
     * Get all alerts with pagination.
     */
    PagedResponseDTO<AlertResponseDTO> getAllAlerts(Pageable pageable);

    /**
     * Get alerts by status.
     */
    PagedResponseDTO<AlertResponseDTO> getAlertsByStatus(AlertStatus status, Pageable pageable);

    /**
     * Get alerts by severity.
     */
    PagedResponseDTO<AlertResponseDTO> getAlertsBySeverity(AlertSeverity severity, Pageable pageable);

    /**
     * Get alerts by status and severity.
     */
    PagedResponseDTO<AlertResponseDTO> getAlertsByStatusAndSeverity(AlertStatus status, AlertSeverity severity, Pageable pageable);

    /**
     * Get alerts by rule ID.
     */
    PagedResponseDTO<AlertResponseDTO> getAlertsByRuleId(Long ruleId, Pageable pageable);

    /**
     * Get alerts by account ID.
     */
    PagedResponseDTO<AlertResponseDTO> getAlertsByAccountId(String accountId, Pageable pageable);

    /**
     * Acknowledge an alert.
     */
    AlertResponseDTO acknowledgeAlert(Long id, String operatorName);

    /**
     * Update alert status.
     */
    AlertResponseDTO updateAlertStatus(Long id, AlertStatusUpdateDTO updateDTO);

    /**
     * Add resolution notes to an alert.
     */
    AlertResponseDTO addResolutionNotes(Long id, String notes);

    /**
     * Get open alerts prioritized by severity.
     */
    List<AlertResponseDTO> getOpenAlertsPrioritized(int limit);

    /**
     * Get alert statistics.
     */
    Map<String, Object> getAlertStatistics();

    /**
     * Count alerts by status.
     */
    long countAlertsByStatus(AlertStatus status);
}

File: frontend/src/services/alertService.js
import apiClient from './api';

export const alertService = {
  // Get all alerts with pagination
  getAll: (page = 0, size = 10) => {
    return apiClient.get(`/alerts?page=${page}&size=${size}`);
  },

  // Get alert by ID
  getById: (id) => {
    return apiClient.get(`/alerts/${id}`);
  },

  // Get alerts by status
  getByStatus: (status, page = 0, size = 10) => {
    return apiClient.get(`/alerts/status/${status}?page=${page}&size=${size}`);
  },

  // Get alerts by severity
  getBySeverity: (severity, page = 0, size = 10) => {
    return apiClient.get(`/alerts/severity/${severity}?page=${page}&size=${size}`);
  },

  // Get prioritized open alerts
  getOpenPrioritized: (limit = 20) => {
    return apiClient.get(`/alerts/open/prioritized?limit=${limit}`);
  },

  // Acknowledge alert
  acknowledge: (id, operatorName = 'Admin') => {
    return apiClient.put(`/alerts/${id}/acknowledge?operatorName=${encodeURIComponent(operatorName)}`);
  },

  // Update alert status
  updateStatus: (id, statusData) => {
    return apiClient.put(`/alerts/${id}/status`, statusData);
  },

  // Get alert statistics
  getStatistics: () => {
    return apiClient.get('/alerts/statistics');
  },

  // Get alert count by status
  getCountByStatus: (status) => {
    return apiClient.get(`/alerts/count/${status}`);
  },
};
